cpp/serve/request_state.h

```diff
@@ -106,10 +106,28 @@ struct DeltaRequestReturn {
   Optional<String> finish_reason;
 };
 
+/*! \brief Request state status. */
+enum class RequestStateStatus : int {
+  kPending = 0,
+  kAlive = 1,
+  kFinished = 2,
+};
+
 class RequestStateNode : public Object {
```

Comment:
Add comment about the overlal request states data structure serving as a tree

Reply:
https://github.com/mlc-ai/mlc-llm/blob/98f3c915b86e0e3d2bdfeb8ad0ee375a27c55838/cpp/serve/request_state.h#L124-L149